### ISSUE TITLE
upgrade to Opal 1.0

### ISSUE THREAD

**Issue by catmando** [hyperstack-org/hyperstack#199]

**Comment by lionelchauvin** [comment]
Did you try to upgrade to Opal 1 ?

For my project, I need to use some gems that use Module#prepend. This feature is only available in Opal 1.0.0

I cloned hyperstack and changed gemspecs in order to upgrade Opal from '>= 0.11.0', '< 0.12.0' to   '>= 1.0.0', '< 1.1.0'  and Opal-rails from 0.9.4 to 1.0.0rc1

Here the problems I encountered:

- opal-sprocket has changed (commit 1f9296596d514bbaca47c32c178893384e7cd0dd) so I have this javascript error:
`Uncaught ReferenceError: Opal is not defined at HTMLScriptElement.onload`

- There is no $method_missing  in Opal.BasicObject.$$prototype. I encountered this errror:
`Uncaught TypeError: Cannot set property '$$p' of undefined at constructor.method_missing_stub`
here: https://github.com/opal/opal/blob/master/opal/corelib/runtime.js#L1301

- After that, react doesn't seems to load properly:
`Uncaught TypeError: Cannot read property 'Component' of undefined  at eval (eval at ./node_modules/react_ujs/react_ujs/src/getConstructor/fromGlobal.js.module.exports`

Any idea to solve these problems ?

**Comment by catmando** [comment]
The first one I do not know but  the Opal team may be able to help:  https://gitter.im/opal/opal

The method_missing is interesting, I would have to look at the back trace and see what is going on, and of course it might be because of the first problem.

I suspect the third problem results from the first two.

**Comment by lionelchauvin** [comment]
I tried to solve this issue and failed again.

this error:
`Uncaught TypeError: Cannot set property '$$p' of undefined at constructor.method_missing_stub`

is due to this line:

https://github.com/hyperstack-org/hyperstack/blob/9e324adccf01f8c7799e92fde8d365fe3026e231/ruby/hyperstack-config/lib/hyperstack/imports.rb#L73

in newer version of opal-rails, opal runtime does not seem to be loaded before "require 'opal'", so
`puts "require \"opal\"; require \"opal\""` fails because "puts" is not yet defined and then it triggers method_missing and crash.

To solve this I replaced puts by a console.log(). Furthermore, I don't like it writes things in production:

```
-          "puts \"#{r}\"; #{r}"
+          Rails.env.production? ? r : "`console.log(\"#{r}\")`; #{r}"
```

After this first bug solved, I had a bug in Opal 1.0.3. It failed to require "opal/corelib/random/MersenneTwister"
It seems that this bug has been fixed in current opal master because MersenneTwister.js code is now in opal/corelib/random/mersenne_twister.rb (see https://github.com/opal/opal/commit/b1074a37150f9b0b53a35f0efa4a1cbddc02ba29). 
I manually changed the code in `.bundle/gems/opal-1.0.3` and wait a newer release of opal.

It still remains a failure when "react_ujs" is required. Opal.load('react_ujs') failed with this error:
`LoadError: cannot load such file -- react_ujs`

I think it is related to this pull request: https://github.com/opal/opal/pull/1970
I don't understand how it was possible to require a node_module in opal 0.11 and why it doesn't work in opal 1.0 anymore.
